Content.Client/Chemistry/UI/ChemMasterWindow.xaml.cs

```diff
@@ -436,42 +399,34 @@ private void BuildPillBufferInfo(ChemMasterBoundUserInterfaceState state)
             };
             bufferHBox.AddChild(bufferVol);
 
-            foreach (var reagent in _pillReagents.Keys)
-            {
-                if (state.BufferReagents.All(x => x.Reagent.Prototype != reagent))
-                    _pillReagents.Remove(reagent);
-            }
-
-            // initialises rowCount to allow for striped rows
-            var rowCount = 0;
             var bufferReagents = state.PillBufferReagents.OrderBy(x => x.Reagent.Prototype);
 
             if (_currentSortMethod == ReagentSortMethod.Amount)
                 bufferReagents = bufferReagents.OrderByDescending(x => x.Quantity);
 
-            if (_currentSortMethod == ReagentSortMethod.Time)
-            {
-                bufferReagents = bufferReagents.OrderByDescending(
-                    x =>
-                    {
-                        var exists = _pillReagents.TryGetValue(x.Reagent.Prototype, out var reagent);
-                        return exists && reagent != null ? reagent.TimeAdded : DateTimeOffset.UtcNow;
-                    });
-            }
+            HandleBuffer(_currentSortMethod == ReagentSortMethod.Time ? state.PillBufferReagents : bufferReagents, true);
+        }
 
-            foreach (var (reagentId, quantity) in bufferReagents)
+        private void HandleBuffer(IEnumerable<ReagentQuantity> reagents, bool pillBuffer)
+        {
+            var rowCount = 0;
+            foreach (var (reagentId, quantity) in reagents)
             {
                 _prototypeManager.TryIndex(reagentId.Prototype, out ReagentPrototype? proto);
 
                 var name = proto?.LocalizedName ?? Loc.GetString("chem-master-window-unknown-reagent-text");
                 var reagentColor = proto?.SubstanceColor ?? default(Color);
-                PillBufferInfo.Children.Add(BuildReagentRow(reagentColor, rowCount++, name, reagentId, quantity, true, true));
-                _pillReagents.TryGetValue(reagentId.Prototype, out var reagentCached);
 
-                var timeAdded = reagentCached?.TimeAdded ?? DateTimeOffset.UtcNow;
-                var cached = new ReagentCached(reagentId, timeAdded, quantity);
-
-                _pillReagents.Add(reagentId.Prototype, cached);
+                if (pillBuffer)
+                {
+                    PillBufferInfo.Children.Add(
+                        BuildReagentRow(reagentColor, rowCount++, name, reagentId, quantity, true, true));
+                }
+                else
+                {
+                    BufferInfo.Children.Add(
+                        BuildReagentRow(reagentColor, rowCount++, name, reagentId, quantity, true, true));
+                }
```

Comment:
```suggestion
                if (pillBuffer)
                    PillBufferInfo.Children.Add(
                        BuildReagentRow(reagentColor, rowCount++, name, reagentId, quantity, true, true));
                else
                    BufferInfo.Children.Add(
                        BuildReagentRow(reagentColor, rowCount++, name, reagentId, quantity, true, true));
```
![faridaiscute](https://github.com/user-attachments/assets/6f46cbcf-a9af-42ca-b098-fabc16787fcc)